addons/loyalty_delivery/tests/test_loyalty_delivery.py

```diff
@@ -18,6 +20,70 @@ def setUpClass(cls):
         cls.product_4 = cls.env['product.product'].create({'name': 'A product to deliver'})
         cls.product_uom_unit = cls.env.ref('uom.product_uom_unit')
 
+    def test_free_delivery_cost_with_ewallet(self):
+        """
+        Test that paying a sale order with an ewallet does not affect when free shipping is
+        applied if the free shipping depends on the amount of the order.
+        """
+
+        # Create a delivery product and service.
+        product_delivery_free = self.env['product.product'].create({
+            'name': 'Test Delivery Product',
+            'type': 'service',
+            'list_price': 10.0,
+            'categ_id': self.env.ref('delivery.product_category_deliveries').id,
+        })
+        free_delivery = self.env['delivery.carrier'].create({
+            'name': 'Test Free Delivery',
+            'product_id': product_delivery_free.id,
+            'free_over': True,
+            'amount': 100,
+        })
+
+        # Create an eWallet Program and its corresponding rewards and coupons.
+        program_ewallet = self.env['loyalty.program'].create({
+            'name': 'eWallet',
+            'program_type': 'ewallet',
+            'reward_ids': [(Command.CREATE, 0, {
```

Comment:
```suggestion
            'reward_ids': [Command.create({
```
You should use the methods and not the constant itself.